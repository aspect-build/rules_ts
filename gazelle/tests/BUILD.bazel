load("@bazel_gazelle//:def.bzl", "gazelle_generation_test")


# A separate generation test for each tests/* test case
[
    gazelle_generation_test(
        name = "%s_test" % test_workspace,
        gazelle_binary = "//:gazelle_js_binary",
        test_data = [
            ":_%s-data" % test_workspace,
        ],
    )
    for test_workspace in [t.replace("/WORKSPACE", "") for t in glob(["*/WORKSPACE"])]
]

# Data for each generation test
[
    filegroup(
        name = "_%s-data" % test_workspace,
        srcs = glob(["%s/**" % test_workspace]),
        testonly = True,
        visibility = ["//visibility:private"],
    )
    for test_workspace in [t.replace("/WORKSPACE", "") for t in glob(["*/WORKSPACE"])]
]
