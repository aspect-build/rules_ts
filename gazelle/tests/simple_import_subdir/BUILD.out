load("@aspect_rules_ts//ts:defs.bzl", "ts_project")

ts_project(
    name = "simple_import_subdir",
    srcs = [
        "main.ts",
        "subdir/lib.ts",
    ],
)